db/sql/prototype.sql

```diff
@@ -0,0 +1,491 @@
+/*
+Functions for testing feasibility of moving different Mathesar pieces to the database. These are
+part of the 'remove SQLAlchemy' project.
+
+There are three involved schemas:
+      __msar:  These functions aren't designed to be used except by other Mathesar functions.
+               Generally need preformatted strings as input, won't do quoting, etc.
+        msar:  These functions are designed to be used more easily. They'll format strings, quote
+               identifiers, and so on.
+  msar_views:  This schema is where internal mathesar views will be stored.
+
+The reason they're so abbreviated is to avoid namespace clashes, and also because making them longer
+would make using them quite tedious, since they're everywhere.
+*/
+
+CREATE SCHEMA IF NOT EXISTS __msar;
+CREATE SCHEMA IF NOT EXISTS msar;
+CREATE SCHEMA IF NOT EXISTS msar_views;
+
+----------------------------------------------------------------------------------------------------
+----------------------------------------------------------------------------------------------------
+-- GENERAL DDL FUNCTIONS
+--
+-- Functions in this section are quite general, and are the basis of the others.
+----------------------------------------------------------------------------------------------------
+----------------------------------------------------------------------------------------------------
+
+CREATE OR REPLACE FUNCTION
+__msar.exec_ddl(command text) RETURNS text AS $$/*
+Execute the given command, returning the command executed.
+
+Not useful for SELECTing from tables. Most useful when you're performing DDL.
+
+Args:
+  command: Raw string that will be executed as a command.
+*/
+BEGIN
+  EXECUTE command;
+  RETURN command;
+END;
+$$ LANGUAGE plpgsql RETURNS NULL ON NULL INPUT;
+
+
+CREATE OR REPLACE FUNCTION
+__msar.exec_ddl(command_template text, arguments variadic anyarray) RETURNS text AS $$/*
+Execute a templated command, returning the command executed.
+
+The template is given in the first argument, and all further arguments are used to fill in the
+template. Not useful for SELECTing from tables. Most useful when you're performing DDL.
+
+Args:
+  command_template: Raw string that will be executed as a command.
+  arguments: arguments that will be used to fill in the template.
+*/
+DECLARE formatted_command TEXT;
+BEGIN
+  formatted_command := format(command_template, VARIADIC arguments);
```

Comment:
That's correct. This helps with sanitization since it lets us parameterize everything we could want to, meaning other pieces of the machine will be able to sanitize things. I.e., most machinery is set up to parameterize and sanitize values passed, but not names. So if we want to have, e.g. `psycopg`'s SQL object help us sanitize strings, we need to reframe names as values. We can't make the assumption that names are safe since they're user-defined, so setting up the function this way will make it easier to sanitize things. (Hopefully.)